test/suite/test_backup04.py

```diff
@@ -76,37 +76,13 @@ def update(self, uri, dsize, upd, rows):
             cursor[simple_key(cursor, i)] = str(i) + ':' + upd * dsize
         cursor.close()
 
-    def take_full_backup(self, dir):
-        # Open up the backup cursor, and copy the files.  Do a full backup.
-        cursor = self.session.open_cursor('backup:', None, None)
-        self.pr('Full backup to ' + dir + ': ')
-        os.mkdir(dir)
-        while True:
-            ret = cursor.next()
-            if ret != 0:
-                break
-            newfile = cursor.get_key()
-            sz = os.path.getsize(newfile)
-            self.pr('Copy from: ' + newfile + ' (' + str(sz) + ') to ' + dir)
-            shutil.copy(newfile, dir)
-        self.assertEqual(ret, wiredtiger.WT_NOTFOUND)
-        cursor.close()
-
     # Take an incremental backup and then truncate/archive the logs.
-    def take_incr_backup(self, dir):
-            config = 'target=("log:")'
-            cursor = self.session.open_cursor('backup:', None, config)
-            while True:
-                ret = cursor.next()
-                if ret != 0:
-                    break
-                newfile = cursor.get_key()
-                sz = os.path.getsize(newfile)
-                self.pr('Copy from: ' + newfile + ' (' + str(sz) + ') to ' + dir)
-                shutil.copy(newfile, dir)
-            self.assertEqual(ret, wiredtiger.WT_NOTFOUND)
-            self.session.truncate('log:', cursor, None, None)
-            cursor.close()
+    def take_incr_backup_and_truncate(self, dir):
+        config = 'target=("log:")'
+        cursor = self.session.open_cursor('backup:', None, config)
+        self.take_full_backup(dir, cursor)
```

Comment:
This function might be abit ambiguous, but not quite sure how to name this @sueloverso any thoughts?

Reply:
The truncate is a specific feature for log-based incremental backup. I would just call it `take_log_incr_backup`. I think it is find to leave the block-based one as `take_incr_backup` because that is the predominant type of backup we test and MongoDB uses.

Reply:
awesome ta!